### ISSUE TITLE
change pin theme after some validation

### ISSUE THREAD

**Issue by gabrielintang** [adar2378/pin_code_fields#56]
Hi thank you for this project. this is great. but I do want to change `pinTheme` (underline color) after some validation. for example I have if it's not error, `activeColor` should be `green` and if some validation its not valid, then `activeColor` should be `red` like this : 
```
PinTheme pinTheme = _isOtpValid
          ? PinTheme(
              activeColor: Color.green,
           )
          : PinTheme(
              activeColor: Color.red,
          );

PinCodeTextField(
    pinTheme: pinTheme,
)
```

I set it with `setState` but `pinTheme` is not updated. do you have any suggestion for this kind of scenario ?

**Comment by adar2378** [comment]
Sorry about the situation. We need to reorganize the code a bit to make it work like that. For a quickfix, you can assign a key to the PinCodeTextField like this
`PinCodeTextField(
    key: ValueKey(DateTime.now()),
)`
It will then update if you're using the error shake animation make sure the StreamController is broadcasting the output,
`errorController = StreamController<ErrorAnimationType>.broadcast();`

**Comment by adar2378** [comment]
ab506c389f9d356848c7ba35a0ee18c51951fcb4 fixed it, use the latest package, version 3.0.1. It should fix it. Thanks for pointing it out :)

**Comment by gabrielintang** [comment]
thanks @adar2378 it's working like charm. save me a lot of time 🙏
